ethereum/extethclient/eth_wallet_client.go

```diff
@@ -63,7 +70,17 @@ type ethClient struct {
 
 // NewEthClient creates and returns our extended ethereum client/wallet. The passed context
 // is only used for creation.
-func NewEthClient(ctx context.Context, ec *ethclient.Client, privKey *ecdsa.PrivateKey) (EthClient, error) {
+func NewEthClient(
+	ctx context.Context,
+	env common.Environment,
+	ec *ethclient.Client,
+	privKey *ecdsa.PrivateKey,
+) (EthClient, error) {
+	err := validateEthClient(ctx, env, ec)
+	if err != nil {
+		return nil, err
+	}
+
```

Comment:
move this below the `ec.ChainID()` line and pass the chain ID into this function instead of the eth client. then you can remove the duplicate call to `ChainID` inside `validateEthClient`